web/modules/custom/rdf_etl/src/Plugin/EtlAdms2ConvertPassInterface.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\rdf_etl\Plugin;
+
+use Drupal\KernelTests\KernelTestBase;
+
+/**
+ * Interface for ADMS v2 conversion pass plugin.
+ *
+ * Such a plugin performs only one specific change/pass in the process of
+ * converting to ADMS v1 data to ADMS v2.01.
+ */
+interface EtlAdms2ConvertPassInterface {
+
+  /**
+   * Testing graph.
+   *
+   * @var string
+   */
+  const TEST_GRAPH = 'http://example.com/graph/test/sink';
+
+  /**
+   * The ADMS v2 asset type.
+   *
+   * @var string
+   */
+  const ASSET = 'http://www.w3.org/ns/dcat#Dataset';
+
+  /**
+   * The ADMS v2 catalog of assets type.
+   *
+   * @var string
+   */
+  const ASSET_CATALOG = 'http://www.w3.org/ns/dcat#Catalog';
+  /**
+   * The ADMS v2 asset distribution type.
+   *
+   * @var string
+   */
+  const ASSET_DISTRIBUTION = 'http://www.w3.org/ns/dcat#Distribution';
+
+  /**
+   * Transforms the triples in the backend.
+   *
+   * @param array $data
+   *   Data received from the process step plugin.
+   */
+  public function convert(array $data): void;
+
+  /**
+   * Performs testing assertions.
+   *
+   * @param \Drupal\KernelTests\KernelTestBase $test
+   *   The testing class.
+   */
+  public function performAssertions(KernelTestBase $test): void;
```

Comment:
Can you explain why this is here? This takes a KernelTestBase class and below there is a getTestngRdfData while these are defined as conversion plugins. How come this is not part of a test or so?

Reply:
See my explanation in the issue ticket.